crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -2284,7 +2281,31 @@ impl<Codec: RawEntryCodec> StateMachine<Codec> {
                 )
                 .await?;
             }
-            EnrichedEntryHeader::Run { .. } | EnrichedEntryHeader::Custom { .. } => {
+            EnrichedEntryHeader::Run { .. } => {
+                let_assert!(Entry::Run(entry) = journal_entry.deserialize_entry_ref::<Codec>()?);
+
+                let name = if entry.name.is_empty() {
+                    Cow::Borrowed("run")
+                } else {
+                    Cow::Owned(format!("run {}", entry.name))
+                };
+
+                let _span = instrumentation::info_invocation_span!(
+                    relation = invocation_metadata
+                        .journal_metadata
+                        .span_context
+                        .as_parent(),
+                    id = invocation_id,
+                    name = name,
```

Comment:
Even better, if you put the logic to read the name straight here, it will execute only if the span is enabled i think (down the line, i think this macro won't execute if the span is not enabled, right?).